src/main/java/com/juniorstart/juniorstart/service/UserProfileService.java

```diff
@@ -0,0 +1,119 @@
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.EnumUtils;
+import org.apache.commons.text.WordUtils;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/** Represents an userProfile service.
+ * @author Rafał maduzia
+ * @version 1.0
+ */
+@Service
+@Slf4j
+public class UserProfileService {
+
+    final private UserProfileRepository userProfileRepository;
+    final private UserDao userRepository;
+
+    public UserProfileService(UserDao userDao, UserProfileRepository userProfileRepository) {
+        this.userProfileRepository = userProfileRepository;
+        this.userRepository = userDao;
+    }
+
+
+    //Ignore This, Future for Next task
+  //  Long userId = new getLogedUserId().userID;
+
+    //TODO GONNA WORK ON ADD USER PROFILE ON NEXT TASK
+    public UserProfile addUserProfile(UserProfile userProfile) {
+
+        //Long userId = new getLogedUserId().userID;
+        //Long tmp = userId;
+        Long userId= 5L;
+
+        Optional<User> foundUser = userRepository.findByPublicId(userId);
+
+        if(foundUser.isPresent()){
+            userProfile.setUser(foundUser.get());
+            userProfile.setPrivateId(foundUser.get().getPrivateId());
+            userProfileRepository.save(userProfile);
+        }
+
+        return userProfile;
+    }
+
+    //TODO GONNA WORK ON UPDATE USER PROFILE ON NEXT TASK
+    public UserProfile updateUserProfile(UserProfile userProfile) {
+
+        Long userId= 5L;
+
+        Optional<UserProfile> foundUser = userProfileRepository.findById(userId);
+
+        if(foundUser.isPresent()){
+            userProfileRepository.save(userProfile);
+        }
+        return userProfile;
+    }
+
+    /** Get a List of UserProfile.
+     * @param technology Technology name you are looking for
+     * @param userRole UserRole(JUNIOR,MENTOR etc) name you are looking for
+     * @return list of UserProfile
+     * @throws ResourceNotFoundException if userRole isn't valid
+     */
+    public List<UserProfile> findByTechnologyAndRole(List<String> technology, List<String> userRole) {
+        List<UserRole> convertedUserRole= validateAndReturnAsEnum(userRole);
+        List<String> convertedTechnology = technology.stream().map(WordUtils::capitalize).collect(Collectors.toList());
+        return userProfileRepository.findByUserTechnology_technologyNameInAndUserRoleIn(convertedTechnology, convertedUserRole);
+    }
+
+    /** Get a List of UserProfile.
+     * @param technology Technology name you are looking for
+     * @return list of UserProfile
+     */
+    public List<UserProfile> findByTechnology(List<String> technology) {
+        List<String> convertedTechnology = technology.stream().map(WordUtils::capitalize).collect(Collectors.toList());
+        return userProfileRepository.findByUserTechnology_technologyNameIn(convertedTechnology);
```

Comment:
źle nazwana metoda userProfileRepository.

Reply:
Nie do końca rozumiem, co masz na myśli ?

Reply:
findByUserTechnologyAndTechnologyNameIn powinna się nazywać

Reply:
Jesteś pewien,  żeby to nadpisywać ? Ta nazwa metoda jest nadana z JPA

Reply:
Jeśli działa jak trzeba to nie zmieniaj :P

Reply:
@rmaduzia jak tak to spoko